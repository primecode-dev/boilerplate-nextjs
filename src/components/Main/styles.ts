import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #171738;
  color: #fff;
  text-align: center;
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 3rem;
`
export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 3.5rem;
  color: #c3c8d9;
  font-weight: bold;
`

export const Description = styled.h2`
  margin-top: 1rem;
  font-size: 2.5rem;
  max-width: 70rem;
  color: #7180b9;
`
