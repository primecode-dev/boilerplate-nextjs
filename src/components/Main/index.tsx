import * as S from './styles'

const Main = ({
  title = 'Boilerplate Next.js',
  description = 'Projeto iniciado com sucesso, agora só customizar seus componentes e criar as páginas'
}) => (
  <S.Wrapper>
    <S.Logo
      src="/assets/images/logo.svg"
      alt="Escrita Primecode com imagem de um diamente"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Main
