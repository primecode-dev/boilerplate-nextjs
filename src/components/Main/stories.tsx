import { withKnobs, text } from '@storybook/addon-knobs'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
}

export const Basic = () => (
  <Main
    title={text('Title', 'Boilerplate Next.js')}
    description={text(
      'Description',
      'Projeto iniciado com sucesso, agora só customizar seus componentes e criar as páginas'
    )}
  />
)
