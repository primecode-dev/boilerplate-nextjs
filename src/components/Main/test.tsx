import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /boilerplate next.js/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#171738' })
    expect(
      screen.getByRole('heading', { name: /boilerplate next.js/i })
    ).toHaveStyle({ color: '#c3c8d9' })
    expect(
      screen.getByRole('heading', { name: /projeto iniciado com sucesso/i })
    ).toHaveStyle({ color: '#7180b9' })
  })
})
