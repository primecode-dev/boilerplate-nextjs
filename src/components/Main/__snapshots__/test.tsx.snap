// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #171738;
  color: #fff;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 3rem;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 3.5rem;
  color: #c3c8d9;
  font-weight: bold;
}

.c3 {
  margin-top: 1rem;
  font-size: 2.5rem;
  max-width: 70rem;
  color: #7180b9;
}

<main
  class="c0"
>
  <img
    alt="Escrita Primecode com imagem de um diamente"
    class="c1"
    src="/assets/images/logo.svg"
  />
  <h1
    class="c2"
  >
    Boilerplate Next.js
  </h1>
  <h2
    class="c3"
  >
    Projeto iniciado com sucesso, agora só customizar seus componentes e criar as páginas
  </h2>
</main>
`
